src/local/shared.py

```diff
@@ -0,0 +1,43 @@
+from collections import defaultdict
+from contextlib import contextmanager
+from logging import getLogger
+from platform import system
+from typing import Any, Dict
+import vdf
+from galaxy.api.types import LocalGameState
+
+log = getLogger(__name__)
+
+
+# Get Windows OS
+
+if system() not in ["Windows", "Darwin"]:
+    raise RuntimeError("OS not supported: " + system())
+
+IS_WIN = system() == "Windows"
+
+
+# VDF stuff
+
+class CaseInsensitiveDict(dict):
```

Comment:
This one is missing an implementation for `get(key, default=None)`. Maybe do a safety-safe there and add it despite you're currently not using it yourself

Reply:
well this class is just copy and pasted for earlier. honestly i dont want to go down the rabbit hole of implementing everything for a dict alternative. if its ever needed for more then it can be added does that sound alr?